Error while building

From a fresh clone of the repo; while running `cargo build` i'm getting:

```
....


error[E0277]: the trait bound `R: ark_std::rand::Rng` is not satisfied
  --> src/secret_sharing/shamir.rs:54:26
   |
54 |         .map(|_| F::rand(rng))
   |                  ------- ^^^ the trait `ark_std::rand::RngCore` is not implemented for `R`
   |                  |
   |                  required by a bound introduced by this call
   |
   = note: required for `R` to implement `ark_std::rand::Rng`
note: required by a bound in `ark_std::UniformRand::rand`
  --> /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ark-std-0.5.0/src/rand_helper.rs:12:16
   |
12 |     fn rand<R: Rng + ?Sized>(rng: &mut R) -> Self;
   |                ^^^ required by this bound in `UniformRand::rand`
help: consider further restricting this bound
   |
39 | pub fn share<R: Rng + ark_std::rand::RngCore>(
   |                     ++++++++++++++++++++++++

error[E0277]: the trait bound `R: ark_std::rand::Rng` is not satisfied
  --> src/secret_sharing/shamir.rs:78:33
   |
78 |                 let x = F::rand(rng);
   |                         ------- ^^^ the trait `ark_std::rand::RngCore` is not implemented for `R`
   |                         |
   |                         required by a bound introduced by this call
   |
   = note: required for `R` to implement `ark_std::rand::Rng`
note: required by a bound in `ark_std::UniformRand::rand`
  --> /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ark-std-0.5.0/src/rand_helper.rs:12:16
   |
12 |     fn rand<R: Rng + ?Sized>(rng: &mut R) -> Self;
   |                ^^^ required by this bound in `UniformRand::rand`
help: consider further restricting this bound
   |
39 | pub fn share<R: Rng + ark_std::rand::RngCore>(
   |                     ++++++++++++++++++++++++

error[E0277]: the trait bound `rand_chacha::rand_core::OsRng: ark_std::rand::Rng` is not satisfied
   --> src/secure_channel/sign.rs:34:32
    |
34  |     let sk = SecretKey::random(&mut OsRng);
    |              ----------------- ^^^^^^^^^^ the trait `ark_std::rand::RngCore` is not implemented for `rand_chacha::rand_core::OsRng`
    |              |
    |              required by a bound introduced by this call
    |
note: there are multiple different versions of crate `rand_core` in the dependency graph
   --> /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rand_core-0.6.4/src/lib.rs:142:1
    |
142 | pub trait RngCore {
    | ^^^^^^^^^^^^^^^^^ this is the required trait
    |
   ::: src/secret_sharing/shamir.rs:21:5
    |
21  | use ark_poly::{Polynomial, univariate::DensePolynomial};
    |     -------- one version of crate `rand_core` is used here, as a dependency of crate `rand`
    |
   ::: src/secret_sharing/vss.rs:23:5
    |
23  | use rand_chacha::rand_core::SeedableRng;
    |     ----------- one version of crate `rand_core` is used here, as a dependency of crate `rand_chacha`
    |
   ::: /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rand_core-0.9.0/src/os.rs:47:1
    |
47  | pub struct OsRng;
    | ---------------- this type doesn't implement the required trait
    |
   ::: /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rand_core-0.6.4/src/os.rs:48:1
    |
48  | pub struct OsRng;
    | ---------------- this type implements the required trait
    |
   ::: /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rand_core-0.9.0/src/block.rs:67:1
    |
67  | pub trait BlockRngCore {
    | ---------------------- this is the found trait
    = note: two types coming from two different versions of the same crate are different types even if they look the same
    = help: you can use `cargo tree` to explore your dependency tree
    = note: required for `rand_chacha::rand_core::OsRng` to implement `ark_std::rand::Rng`
note: required by a bound in `SecretKey::random`
   --> /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libsecp256k1-0.7.1/src/lib.rs:430:22
    |
430 |     pub fn random<R: Rng>(rng: &mut R) -> SecretKey {
    |                      ^^^ required by this bound in `SecretKey::random`

error[E0277]: the trait bound `rand_chacha::rand_core::OsRng: ark_std::rand::Rng` is not satisfied
   --> src/secure_channel/encrypt.rs:24:32
    |
24  |     let sk = SecretKey::random(&mut OsRng);
    |              ----------------- ^^^^^^^^^^ the trait `ark_std::rand::RngCore` is not implemented for `rand_chacha::rand_core::OsRng`
    |              |
    |              required by a bound introduced by this call
    |
note: there are multiple different versions of crate `rand_core` in the dependency graph
   --> /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rand_core-0.6.4/src/lib.rs:142:1
    |
142 | pub trait RngCore {
    | ^^^^^^^^^^^^^^^^^ this is the required trait
    |
   ::: src/secret_sharing/shamir.rs:21:5
    |
21  | use ark_poly::{Polynomial, univariate::DensePolynomial};
    |     -------- one version of crate `rand_core` is used here, as a dependency of crate `rand`
    |
   ::: src/secret_sharing/vss.rs:23:5
    |
23  | use rand_chacha::rand_core::SeedableRng;
    |     ----------- one version of crate `rand_core` is used here, as a dependency of crate `rand_chacha`
    |
   ::: /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rand_core-0.9.0/src/os.rs:47:1
    |
47  | pub struct OsRng;
    | ---------------- this type doesn't implement the required trait
    |
   ::: /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rand_core-0.6.4/src/os.rs:48:1
    |
48  | pub struct OsRng;
    | ---------------- this type implements the required trait
    |
   ::: /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rand_core-0.9.0/src/block.rs:67:1
    |
67  | pub trait BlockRngCore {
    | ---------------------- this is the found trait
    = note: two types coming from two different versions of the same crate are different types even if they look the same
    = help: you can use `cargo tree` to explore your dependency tree
    = note: required for `rand_chacha::rand_core::OsRng` to implement `ark_std::rand::Rng`
note: required by a bound in `SecretKey::random`
   --> /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libsecp256k1-0.7.1/src/lib.rs:430:22
    |
430 |     pub fn random<R: Rng>(rng: &mut R) -> SecretKey {
    |                      ^^^ required by this bound in `SecretKey::random`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `derec-crypto` (lib) due to 4 previous errors
warning: build failed, waiting for other jobs to finish...
➜  cryptography git:(main) git status
On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean
```

System:
- Debian 12

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -14,26 +14,26 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-serde = { version = "*", features = ["derive"] }
-serde_json = "*"
-sha2 = "*"
-aead = "*"
-aes = "*"
-aes-gcm = "*"
-anyhow = "*"
-rand = "*"
-rand_chacha = "*"
-ark-std = "*"
-ark-ec = "*"
-ark-ff = "*"
-ark-poly = "*"
-ark-serialize = { version = "*", default-features = true }
-ark-bls12-381 = "*"
+serde = { version = "1.0.219", features = ["derive"] }
+serde_json = "1.0.140"
+sha2 = "0.10.9"
+aead = "0.5.2"
+aes = "0.8.4"
+aes-gcm = "0.10.3"
+anyhow = "1.0.98"
+rand = "0.8.5"
+rand_chacha = "0.3.1"
+ark-std = "0.5.0"
+ark-ec = "0.5.0"
+ark-ff = "0.5.0"
+ark-poly = "0.5.0"
+ark-serialize = { version = "0.5.0", default-features = true }
+ark-bls12-381 = "0.5.0"
 pgp = "0.9.0"
-smallvec = "*"
-base64 = "*"
-hex = "*"
-thiserror = "*"
-pem = "*"
-ecies = {version = "*", features = ["std"]}
-libsecp256k1 = "*"
+smallvec = "1.15.0"
+base64 = "0.22.1"
+hex = "0.4.3"
+thiserror = "1.0.69"
+pem = "3.0.5"
+ecies = { version = "0.2.9", features = ["std"] }
+libsecp256k1 = "0.7.2"

